pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -156,16 +156,48 @@ func (m *nodeIDToNameMap) remove(nodeID string) {
 	delete(m.items, nodeID)
 }
 
+// Map of volume ID to volume name.
+// Key is the volume ID and value is the volume name.
+// The methods to add, remove and get entries from the map in a threadsafe
+// manner are defined.
+type volumeIDToNameMap struct {
+	*sync.RWMutex
+	items map[string]string
+}
+
+// Adds an entry to volumeNameToIDMap in a thread safe manner.
+func (m *volumeIDToNameMap) add(volumeID, volumeName string) {
+	m.Lock()
+	defer m.Unlock()
+	m.items[volumeID] = volumeName
+}
+
+// Removes a volume ID from volumeNameToIDMap in a thread safe manner.
+func (m *volumeIDToNameMap) remove(volumeID string) {
+	m.Lock()
+	defer m.Unlock()
+	delete(m.items, volumeID)
+}
+
+// Returns the volume ID corresponding to volumeName.
+func (m *volumeIDToNameMap) get(volumeID string) (string, bool) {
+	m.RLock()
+	defer m.RUnlock()
+	volumeName, found := m.items[volumeID]
+	return volumeName, found
+}
+
 // K8sOrchestrator defines set of properties specific to K8s.
 type K8sOrchestrator struct {
-	supervisorFSS      FSSConfigMapInfo
-	internalFSS        FSSConfigMapInfo
-	informerManager    *k8s.InformerManager
-	clusterFlavor      cnstypes.CnsClusterFlavor
-	volumeIDToPvcMap   *volumeIDToPvcMap
-	volumeIDToNodesMap *volumeIDToNodesMap
-	nodeIDToNameMap    *nodeIDToNameMap
-	k8sClient          clientset.Interface
+	supervisorFSS        FSSConfigMapInfo
+	internalFSS          FSSConfigMapInfo
+	informerManager      *k8s.InformerManager
+	clusterFlavor        cnstypes.CnsClusterFlavor
+	volumeIDToPvcMap     *volumeIDToPvcMap
+	volumeNameToNodesMap *volumeNameToNodesMap
```

Comment:
can we keep volumeNameToNodesMap and volumeIDToNameMap together and also add comments to know which feature is using this field.

Reply:
done